editor/src/components/common/notices.tsx

```diff
@@ -17,14 +19,16 @@ export interface Notice {
   message: React.ReactChild
   persistent?: boolean
   level?: NoticeLevel
+  id?: string
```

Comment:
probably not the best pattern but I was unclear whether it's better to make the ID non-optional and force each caller to provide it (but then how to enforce uniqueness), or to make something like a InnerNotice type (without ID), and an OuterNotice prop (with ID).

Reply:
It seems like ID should be made non-optional. Using a random (v4) UUID is enough to enforce uniqueness since the likelihood of clash is so small.

Reply:
Can you suggest a way to do this that addresses my comment from above? Issue is unless there's a type that enforces or correlates with uniqueness, we rely on _anywhere in our codebase_ to actually _use_ UUID.